### ISSUE TITLE
vm: implement stack

### ISSUE THREAD

**Issue by obscuren** [obscuren/tinyvm#2]

**Comment by kehiy** [comment]
@obscuren I can work on that, any details of where this stack must be used? so after implementing it I can use it where we need it and make a PR.

**Comment by obscuren** [comment]
This was 2016… I honestly can’t remember. 

Are you using this for anything in specific? Do you require a functioning stack?

**Comment by kehiy** [comment]
@obscuren Oh, that's logical I can understand. I'm reading the code and working on it to learn. I'm working on VMs and I'm interested in developing them. I found this to be very straightforward and nice good. I'm just trying to finish the unfinished parts.

**Comment by kehiy** [comment]
I'll try to read the code to find out where we need a stack.

**Comment by kehiy** [comment]
@obscuren I think this was planned for Push(s) and Pop OP codes. Am I right? If you think it's correct, let me know. So I can finish this.

Also, what is the datatype of our stack items? can it be `uint256` the same as EVM?

**Comment by obscuren** [comment]
No, push and pop only support 32bits. See https://github.com/obscuren/tinyvm?tab=readme-ov-file#instruction-encoding

**Comment by kehiy** [comment]
Got it. in next PR I'll add it.
